processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -181,11 +181,45 @@ public BeanMappingMethod build() {
             if ( factoryMethod == null ) {
                 if ( selectionParameters != null && selectionParameters.getResultType() != null ) {
                     resultType = ctx.getTypeFactory().getType( selectionParameters.getResultType() );
-                    if ( !resultType.isAssignableTo( method.getResultType() ) ) {
+                    if ( resultType.isInterface() ) {
                         ctx.getMessager().printMessage(
                             method.getExecutable(),
                             beanMappingPrism.mirror,
-                            Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
+                            Message.BEANMAPPING_INTERFACE,
+                            resultType,
+                            method.getResultType()
+                        );
+                    }
+                    else if ( !resultType.isAssignableTo( method.getResultType() ) ) {
+                        ctx.getMessager().printMessage(
+                        method.getExecutable(),
+                        beanMappingPrism.mirror,
+                        Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
+                        );
+                    }
+                    else if ( resultType.isAbstract() ) {
```

Comment:
It will be done at the beginning and it abstract and interface will be considered the same